docs/dev-concepts/participant-authentication.md

```diff
@@ -0,0 +1,76 @@
+
+<!--In the message encryption flows I show the sender signing the message with their prekey (I should specify private prekey). For the message decryption flow, I show the recipient verifying that message sender using the sender's public key from the message header - specifically, is this the public prekey? https://xmtp.org/docs/dev-concepts/invitation-and-message-encryption#message-encryption-->
+
+An XMTP identity may have multiple private prekeys. A client must be able to access every private prekey used by an XMTP identity to decrypt historical messages.
+
+There is always only one public prekey for an XMTP identity, which is the most current prekey.
+
+## Key generation and retrieval flow
+
+The following diagrams show how a client generates and retrieves identity keys and prekeys. The diagrams also show details of how a client encrypts and stores keys.
+
+Encrypted keys generated for a blockchain account are accessible by only the client apps authorized by the user (controller of the blockchain account). By design, XMTP and the XMTP network cannot access these encrypted keys. <!--I think we should remove this "By design..." sentence. The message in the previous sentence is clearer without it. By mentioning two things that cannot access the keys only opens up the question of what else can't access the keys. Also, what do we mean by XMTP? I believe we mean XMTP Labs.-->
+
+To start the flow, a user accesses an app built with XMTP and connects the app to their desired blockchain account using their wallet app. The app then creates a client.
+
+The client checks the XMTP network for keys associated with the account.
+
+If the keys don't exist, [the client generates them](#key-generation-flow).
+
+If the keys exist, [the client retrieves them](#key-retrieval-flow).
+
+### Key generation flow
+
+If the client checks the XMTP network for keys associated with the account and the keys don't exist, the client generates them according to this flow.
+
+![Flowchart shows how a message API client generates and stores identity keys and prekeys needed to support a user's messaging session](img/key-generation-flow.png)<!--source file: https://www.figma.com/file/77ToMB4T16NiLwJjIp7dU1/diagrams?node-id=47%3A755-->
+
+### Key retrieval flow
+
+If the client checks the XMTP network for keys associated with the account and the keys exist, the client can retrieve them according to this flow.
+
+The retrieval flow is not dictated by the XMTP client protocol. After the client has generated the keys, client apps can determine how to best retrieve and store keys. <!--something like this to define that the retrieval flow is not a part of the protocol?-->
```

Comment:
It's both storage and retrieval that I'd say are not part of the protocol. But maybe we don't really need to get into this here. It's not in the spec, so that may be clear enough.

Reply:
Got it. Thank you, Martin - removing this paragraph.